torchgeo/datasets/geo.py

```diff
@@ -487,6 +487,7 @@ def _merge_files(
                 indexes=band_indexes,
                 out_shape=out_shape,
                 window=from_bounds(*bounds, src.transform),
+                boundless=True,
             )
         else:
             dest, _ = rasterio.merge.merge(
```

Comment:
`merge` has no `boundless` parameter, but doesn't need one. I confirmed that it properly samples outside the bounds of the file and uses nodata pixels for padding.